src/extras.cpp

```diff
@@ -23,7 +23,7 @@ std::vector<uint32_t> mblockread(uint32_t const& addr, uint32_t const& count)
   std::vector<uint32_t> data(count, 0);
 
   if (memhub_read(memsvc, addr, count, data.data()) != 0) {
-    std;;stringstream errmsg;
+    std::stringstream errmsg;
     errmsg << "blockread memsvc error: " << memsvc_get_last_error(memsvc);
     LOG4CPLUS_ERROR(logger, errmsg.str());
     raise std::runtime_error(errmsg.str); 
```

Comment:
Is it compiling? Isn't `raise` Python and `throw` C++ or did you include a specific header? See lines 53, 77, ,98 and 118 too.